src/library/dlgtrackinfo.cpp

```diff
@@ -606,16 +606,32 @@ void DlgTrackInfo::slotKeyTextChanged() {
 }
 
 void DlgTrackInfo::slotImportMetadataFromFile() {
-    if (m_pLoadedTrack) {
-        // Allocate a temporary track object for reading the metadata.
-        // We cannot reuse m_pLoadedTrack, because it might already been
-        // modified and we want to read fresh metadata directly from the
-        // file. Otherwise the changes in m_pLoadedTrack would be lost.
-        TrackPointer pTrack = SoundSourceProxy::importTemporaryTrack(
-                m_pLoadedTrack->getFileAccess());
-        DEBUG_ASSERT(pTrack);
-        populateFields(*pTrack);
+    if (!m_pLoadedTrack) {
+        return;
+    }
+    mixxx::TrackMetadata trackMetadata;
+    QImage coverImage;
+    const auto [importResult, metadataSynchronized] =
+            SoundSourceProxy(m_pLoadedTrack)
+                    .importTrackMetadataAndCoverImage(
+                            &trackMetadata, &coverImage);
+    if (importResult != mixxx::MetadataSource::ImportResult::Succeeded) {
```

Comment:
Should this print a warning? Under which conditions does this fail?

Reply:
It didn't print a warning beforehand. If it fails (missing or corrupt file) a warning has already been logged.

I don't want to introduce a message box here, since we don't know why it failed. If it fails nothing happens. Errors should better be displayed in a common info pane, because this might also happen when batch importing tags where a message box is inappropriate.